modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-shipment-test/src/testIntegration/java/com/liferay/headless/commerce/admin/shipment/resource/v1_0/test/ShipmentResourceTest.java

```diff
@@ -259,6 +274,70 @@ private Shipment _addShipment(String externalReferenceCode)
 		return _toShipment(_commerceShipment);
 	}
 
+	private ShippingAddress _randomShippingAddress() throws Exception {
+		CommerceAddress commerceAddress = _commerceOrder.getShippingAddress();
+
+		Country country = commerceAddress.getCountry();
+		Region region = commerceAddress.getRegion();
+
+		return new ShippingAddress() {
+			{
+				city = RandomTestUtil.randomString();
+				countryISOCode = country.getA2();
+				description = RandomTestUtil.randomString();
+				externalReferenceCode = RandomTestUtil.randomString();
+				id = RandomTestUtil.randomLong();
+				latitude = RandomTestUtil.randomDouble();
+				longitude = RandomTestUtil.randomDouble();
+				name = RandomTestUtil.randomString();
+				phoneNumber = RandomTestUtil.randomString();
+				regionISOCode = region.getRegionCode();
+				street1 = RandomTestUtil.randomString();
+				street2 = RandomTestUtil.randomString();
+				street3 = RandomTestUtil.randomString();
+				zip = RandomTestUtil.randomString();
+			}
+		};
+	}
+
+	private void _testPostShipmentWithMoreExternalReferenceCodes()
+		throws Exception {
+
+		Shipment randomShipment = randomShipment();
+
+		ShippingAddress randomShippingAddress = _randomShippingAddress();
+
+		randomShippingAddress.setId(0L);
+
+		long orderId = randomShipment.getOrderId();
+
+		randomShipment.setOrderId(0L);
+		randomShipment.setOrderExternalReferenceCode(
+			_commerceOrder.getExternalReferenceCode());
+		randomShipment.setShippingAddress(randomShippingAddress);
+		randomShipment.setShippingAddressId(0L);
+
+		Shipment postShipment = shipmentResource.postShipment(randomShipment);
+
+		randomShipment.setOrderId(orderId);
+
+		ShippingAddress postShippingAddress = postShipment.getShippingAddress();
+
+		ShippingAddress getShippingAddress = _toShippingAddress(
+			_commerceAddressService.getCommerceAddress(
+				postShippingAddress.getId()));
+
+		randomShippingAddress.setId(getShippingAddress.getId());
+		randomShipment.setShippingAddressId(getShippingAddress.getId());
+
+		Assert.assertEquals(
+			randomShippingAddress.getExternalReferenceCode(),
+			getShippingAddress.getExternalReferenceCode());
```

Comment:
because here we can do the check with the postShippingAddress

Reply:
@smottal I did that at first, but I thought this is better, I'm going to change it back

Reply:
@smottal I think it is better with get because than we are just comparing post and the object we have provided earlier to post. By using the id we can make sure that there was also an ID generated that belongs to the exact address with the correct ERC.

Reply:
@brbalazs ok, go with the get. let me know when done that I want to check the names. thx